### PR TITLE
FastReseg for segmentation error evaluation

### PR DESCRIPTION
**Pull request by lidanwu**
This PR added a blog on using FastReseg for cell segmentation evaluation and correction with a focus on preparing the required inputs from AtoMx exported basic files. More details on tuning parameters of FastReseg is discussed in its own tutorial inside the package. 
Local rendering of the blog looks correct. 
image
image

### CODE REVIEW

File: posts/segmentation-error-evaluation/index.qmd [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+4. FastReseg further identifies the spatially connected groups of transcripts with low goodness-of-fit within incorrectly segmented cells. 
+5. A set of heuristic rules on neighborhood cell typing and transcript number are then applied to the identified transcript groups to decide on the re-segmentation actions, like merging, splitting and trimming. The re-segmented results show no significant spatial dependency on transcript score of individual cells, suggesting the successful correction of poorly segmented cells.
+
+
+![](figures/Fig1-FastReseg-workflow.png)
+
+
+You can find the FastReseg package [here](https://github.com/Nanostring-Biostats/FastReseg/tree/main){target="_blank"}. See the corresponding [tutorial](https://github.com/Nanostring-Biostats/FastReseg/blob/main/vignettes/tutorial.html){target="_blank"} inside the package for more details.
```
Comment (eveilyeverafter) [inline posts/segmentation-error-evaluation/index.qmd]: These links do not work for me.
image

File: posts/segmentation-error-evaluation/index.qmd [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+
+* `refProfiles`: a gene-by-cluster matrix of cluster-specific expression profiles; default = `NULL` to use external cluster assignments.
+
+* `transDF_fileInfo`: a data.frame with each row for each individual file of per-FOV transcript data.frame, columns include the file path of per FOV transcript data.frame file, annotation columns like `slide` and `fov` to be used as prefix when creating unique cell_ID across entire dataset.
+   + when `NULL`, use the transcript data.frame `transcript_df` directly.
+
+
+This post will show you how to prepare your inputs if you have data assembled in the structure used by the Technology Access Program [TAP](https://nanostring.com/products/cosmx-spatial-molecular-imager/technology-access-program/); similar outputs are available from the AtoMx^TM^ Spatial Informatics Portal (SIP). You can download an example public dataset from [here](https://nanostring.com/products/cosmx-spatial-molecular-imager/ffpe-dataset/cosmx-smi-human-pancreas-ffpe-dataset/){target="_blank"}. The [ReadMe](./assets/Pancreas-CosMx-ReadMe.html) associated with this example data set on pancreas shows the data structures of each file used in this post.
```
Comment (eveilyeverafter) [inline posts/segmentation-error-evaluation/index.qmd]: Technology Access Program (TAP)

File: posts/segmentation-error-evaluation/index.qmd [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+
+
+Like other items in our [CosMx Analysis Scratch Space](https://nanostring-biostats.github.io/CosMx-Analysis-Scratch-Space/about.html){target="_blank"},
+the usual [caveats](https://nanostring-biostats.github.io/CosMx-Analysis-Scratch-Space/about.html){target="_blank"} and [license](https://nanostring-biostats.github.io/CosMx-Analysis-Scratch-Space/license.html){target="_blank"} applies.
+
+
+
+# Prepare inputs from basic data files {#sec-loading-flatFiles}
+
+Here we start from the basic data files exported from AtoMx^TM^ SIP.
```
Comment (eveilyeverafter) [inline posts/segmentation-error-evaluation/index.qmd]: Typing `AtoMx&#8482;` works fine for me. Did you forget the ; when you tried?

File: posts/segmentation-error-evaluation/index.qmd [hunk 4]
```diff
@@ -0,0 +1,431 @@
+---
+title: "Evaluating Cell Segmentation Error based on Transcripional Spatial Profiles using FastReseg"
+author:
+  - name: Lidan Wu
+    affiliations:
+      - ref: nstg
+      - ref: lidanwu
+affiliations:
+  - id: nstg
+    name: NanoString Technologies, Inc.
```
Comment (eveilyeverafter) [inline posts/segmentation-error-evaluation/index.qmd]: NanoString, a Bruker Company

(we need to switch other post too so we can update this now or later, whichever you like)

File: posts/segmentation-error-evaluation/index.qmd [hunk 5]
```diff
@@ -0,0 +1,431 @@
...
+                   fov = fovId, 
+                   stage_X = 0, 
+                   stage_Y = 0)
+  return(df)
+})
+
+transDF_fileInfo <- do.call(rbind, transDF_fileInfo)
+```
+
+`FastReseg` evaluates segmentation in physical space and thus it's preferred to convert all 3D coordinates into same unit. The relevant default parameters of `FastReseg` are set with respect to micrometer in coordinate unit. According to the  [ReadMe](./assets/Pancreas-CosMx-ReadMe.html), the pixel size for this data set is 0.12028 um per pixel and the z step size is 0.8um per z slice. We would pass this information to `FastReseg` functions to do the coordinate conversion. If your data is already in micrometer unit, you can use `1` for pixel size and z step to disable the conversion.
```
Comment (eveilyeverafter) [inline posts/segmentation-error-evaluation/index.qmd]: Throughout, you can change um to SI units µm. 

Find/Replace " um " with "&micro;m". 

You can also do LaTeX math mode with $\mu$m but that will increase the line spacing a little.

Comment (eveilyeverafter) [reply posts/segmentation-error-evaluation/index.qmd]: Note: github changed my symbol to the micron symbol. Concatenate & + micro + ; (no spaces, and not plus signs).

File: posts/segmentation-error-evaluation/index.qmd [hunk 6]
```diff
@@ -0,0 +1,431 @@
...
+date-modified: last-modified
+categories: [segmentation, algorithms]
+image: figures/Fig1-FastReseg-workflow.png
+description: "FastReseg algorithm scores individual transcripts for the goodness-of-fit within their respective cells based on the probability of each gene belonging to each cell type and the spatial dependency of transcript score. FastReseg can flag cells with putative cell segmentation errors and perform corrections rapidly. "
+draft: false
+---
+
+# Introduction 
+
+Accurate cell segmentation that assigns transcripts to cell locations is critical to data quality of spatial transcriptomics assays, but very challenging for tissue sections where cells are tightly packaged with shared, 3D boundaries and uneven morphology staining.
```
Comment (eveilyeverafter) [inline posts/segmentation-error-evaluation/index.qmd]: I think the introduction needs a little more unpacking. Can you provide a few examples to help orient the reader why proper segmentation matters. This need not be a concrete example but a little more context will help the non-experts.

File: posts/segmentation-error-evaluation/index.qmd [hunk 7]
```diff
@@ -0,0 +1,431 @@
...
+
+You can see that cells with large `lrtest_nlog10P` value exhibited strong spatial dependency in their transcript score profiles under their best fitted cell type (i.e. `score_tLLR_maxCellType`). Below we would zoom in to one of the flagged cells (`c_1_51_1211` with `lrtest_nlog10P = 49.82`) and visualize its transcript score profiles in 3D. 
+```{ojs}
+//| echo: false
+data = FileAttachment("assets/chosenCell_flagged_transDF.csv").csv({ typed: true })
+```
+```{ojs}
+//| echo: false
+//| panel: input
+viewof colorColumn = Inputs.select(
```
Comment (eveilyeverafter) [inline posts/segmentation-error-evaluation/index.qmd:275]: Great plot!

File: posts/segmentation-error-evaluation/index.qmd [hunk 8]
```diff
@@ -0,0 +1,431 @@
...
+  cutoff_spatialMerge = 0.5, # spatial constraint cutoff for a valid merge event
+  
+  path_to_output = outDir_full,
+  save_intermediates = TRUE, # flag to return and write intermediate results to disk
+  return_perCellData = TRUE, # flag to return per cell level outputs from updated segmentation 
+  combine_extra = FALSE # flag to include trimmed and extracellular transcripts in the exported `updated_transDF.csv` files 
+)
+```
+
+There are several parameters one can adjust to tune the identification of wrongly segmented transcript groups and the rules used for cell segmentation correction. Please refer to [FastReseg tutorial](https://github.com/Nanostring-Biostats/FastReseg/blob/main/vignettes/tutorial.html){target="_blank"}, `Modular functions for individual tasks` section, for more details.
```
Comment (eveilyeverafter) [inline posts/segmentation-error-evaluation/index.qmd]: Could you add a few-sentence conclusion section to wrap up the post?

**Review by eveilyeverafter — APPROVED** [review]
Looks great. I have a few minor comments that I'd like to see changed. Feel free to resolve as appropriate. I don't need to review again.

File: posts/segmentation-error-evaluation/index.qmd (hunk 1 above)
Comment (lidanwu) [reply posts/segmentation-error-evaluation/index.qmd]: Sorry that I just changed the name of default branch to `master` for public release. I'll update it and add you to the repository.

File: posts/segmentation-error-evaluation/index.qmd (hunk 3 above)
Comment (lidanwu) [reply posts/segmentation-error-evaluation/index.qmd]: Oh great, it doesn't work when I rendered it from my end, although Claire's blog look fine on my instance. Let me check it back to `&#8482` and you can let me know if it looks right from your end.

Comment (lidanwu) [reply posts/segmentation-error-evaluation/index.qmd]: ah I was missing `;`

File: posts/segmentation-error-evaluation/index.qmd (hunk 4 above)
Comment (lidanwu) [reply posts/segmentation-error-evaluation/index.qmd]: Is "NanoString, a Bruker Company" the exact words? Does it need "Inc." or other stuffs?

File: posts/segmentation-error-evaluation/index.qmd (hunk 5 above)
Comment (lidanwu) [reply posts/segmentation-error-evaluation/index.qmd]: You mean use `&microm` or micrometer?

**Comment by lidanwu** [comment]
@eveilyeverafter I have edited the posts based on your comments. Please take a look again and let me know if further changes needed.

**Comment by lidanwu** [comment]
@eveilyeverafter , could you please take a look at it again? If it looks good to you. Please feel free to merge this branch.

**Comment by eveilyeverafter** [comment]
I'll take a look.

**Comment by eveilyeverafter** [comment]
lgtm

**Pull request merged**